castra/core.py

```diff
@@ -154,9 +154,9 @@ def extend(self, df):
         if len(self.partitions) and df.index[0] < self.partitions.index[0]:
             if is_trivial_index(df.index):
                 df = df.copy()
-                new_index = pd.Index(range(self.partitions.index[0] + 1,
-                                           self.partitions.index[0] + 1 + len(df)),
-                                    name = df.index.name)
+                start = self.partitions.index[-1] + 1
+                new_index = pd.Index(range(start, start + len(df)),
```

Comment:
should this be `arange`? in the case of large chunks it'll be much more memory efficient